Update collections backend API

tw4l opened the pull request:
Fixes #626 

- Rework collections to store references to crawl_ids rather than the other way around
- Add API endpoints to update and rename collections, as well to add/remove a crawl_id from a collection
- Add collection names to get/list crawl API endpoints
- Remove crawl from all collections when the crawl is deleted

tw4l committed:
Re-implement collections, storing crawl_ids in collection

tw4l committed:
Return collections for crawl endpoints and filter on coll name

tw4l committed:
Remove crawl from all collections when deleted

ikreymer commented on backend/btrixcloud/colls.py:
```diff
@@ -106,83 +178,49 @@ async def list_collections(
+        for crawl_id in coll.crawl_ids:
+            org = await self.orgs.get_org_by_id(oid)
+            crawl = await self.crawls.get_crawl(crawl_id, org)
+            collection_crawls[crawl_id] = crawl.resources
```
We actually want to return a flat array of `resources` objects here, so that this can conform with the 'MultiWACZ' Frictionless Data-based spec, where each resource is a WACZ file from the crawl.
It looks like frictionless requires the resources list to be flat: https://specs.frictionlessdata.io/data-package/#resource-information
Depending on how the spec evolves, the result could be:
```
[
 {"name": "<crawl id>/<filename.wacz>
  "path": "<presigned url>
 }
 ]
 ```
 or maybe:
 ```
 [
 {
  "crawl": "<crawl id>"
  "name": "<filename.wacz>"
  "path": "<presigned url>
  }]
 ```

ikreymer replied:
Or, we keep this as is, and add a separate `/collections/<name>/replay.json` endpoint, like we do for crawls?

tw4l replied:
Ah okay, that makes sense! I was missing the context for this. Happy to make this a flat array - I don't know that we have a use case for this endpoint as a separate entity to warrant a separate replay.json endpoint?

tw4l replied:
Modified so that the result looks like:

```
{"resources":
[
  {
      "name":"84e35263-3bfd-4248-9b75-baf49c8e73ff/20230413163255245-manual-20230413163208-abb695ed-1a4.wacz",
      "path":"/data/84e35263-3bfd-4248-9b75-baf49c8e73ff/20230413163255245-manual-20230413163208-abb695ed-1a4.wacz?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=ADMIN%2F20230413%2F%2Fs3%2Faws4_request&X-Amz-Date=20230413T190612Z&X-Amz-Expires=3600&X-Amz-SignedHeaders=host&X-Amz-Signature=6b474aa09c37baac6a7b56a8c7e97371d68cbd27dd8b640d83b7489add4c1de3","hash":"637feafabe5bef0dbafcfd2b2e8dae8b07d3be9a85b966879f504758123c465a",
      "size":174677,
      "crawlId":"manual-20230413163208-abb695ed-1a4"
  }
]}
```

ikreymer replied:
Nice, that works! Yeah, can revisit separate endpoint if needed for some other reason

tw4l committed:
Revert get_collection_crawls to flat array of resources

tw4l committed:
Fix list endpoint

tw4l committed:
Add crawlId to collection resource list

tw4l committed:
Fix and extend collection backend tests

Add tests for get collection and list collections endpoints, and modify
existing tests to account for backend changes.

tw4l committed:
Rename Collection.crawl_ids to crawlIds for consistency

tw4l commented:
@ikreymer this is now ready for re-review
